ddR/R/pdriver.R

```diff
@@ -25,63 +27,74 @@ setClass("ParallelddR", contains="ddRDriver")
 #' @export 
 # Exported Driver
 parallel <- new("ParallelddR",DListClass = "ParallelObj",DFrameClass = "ParallelObj",DArrayClass = "ParallelObj",backendName = "parallel")
+
+windows <- .Platform$OS.type == "windows"
+
 # Driver for the parallel package. Parallel is also the default backend.
-ddR.env$driver <- parallel
+#ddR.env$driver <- parallel
 
-#Set environment and default number of cores to total no. of cores (but one on windows)
-#Note that DetectCores() can return NA. If it's windows we use SNOW by default
+# TODO Clark: The init method modifies this environment variable.
+# It may be simpler to return this from init.
 parallel.ddR.env <- new.env(emptyenv())
-parallel.ddR.env$cores <- parallel::detectCores(all.tests=TRUE, logical=FALSE) 
-parallel.ddR.env$clusterType <- "FORK"
-parallel.ddR.env$snowCluster <- NULL
-if(is.na(parallel.ddR.env$cores)){ parallel.ddR.env$cores <- 1 }
-if((.Platform$OS.type == "windows")) {parallel.ddR.env$clusterType <- "PSOCK"}
-
-#Function to initialize SNOW on windows
-initializeSnowOnWindows<-function(){
-  cl <- parallel::makeCluster(getOption("cl.cores", parallel.ddR.env$cores))
-  parallel.ddR.env$snowCluster <- cl
-  parallel.ddR.env$clusterType <- "PSOCK"
-}
 
 # Initialize the no. of cores in parallel backend
-# By default we use the FORK method of parallel which works only on UNIx environments. The "PSOCK" method requires SNOW but works on all OSes.
-#' @param executors Number of cores to run with.
+# The FORK method of parallel works only on UNIX environments. The "PSOCK"
+# method requires SNOW but works on all OSes.
+#
+#' @param executors Number of cores to run with, or "all" to use all
+#'      available cores
 #' @param type If "FORK", will use UNIX fork() method. If "PSOCK", will use SNOW method.
+#' @param ... Additional arguments to \link[parallel]{makeCluster}
 #' @describeIn init Initialization for parallel
-setMethod("init","ParallelddR",
-  function(x, executors=NULL, type= "FORK", ...){
-    if(!is.null(executors)){
-    if(!((is.numeric(executors) || is.integer(executors)) && floor(executors)==executors && executors>=0)) stop("Argument 'executors' should be a non-negative integral value")
+setMethod("init", "ParallelddR",
+function(driver, executors = "all",
+         type = ifelse(windows, "PSOCK", "FORK"), ...){
+
+    # Normalize executors to positive integer
+    if(executors == "all"){
+        # NIT- Figure out how to suppress this message:
```

Comment:
Was unsuccessful trying [various things](https://github.com/clarkfitzg/junkyard/blob/master/cat_output.R) with `capture.output` and `sink` to suppress stderr message from system call to `cat`. Looking at the code for `parallel::detectCores` it should be ok to just use default `all.tests=FALSE`.